sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1054,6 +1059,9 @@ static int sof_card_dai_links_create(struct device *dev,
 		if (!codec_name)
 			return -ENOMEM;
 
+		if (!strcmp(info->dai_name, "snd-soc-dummy-dai"))
+			codec_name = "snd-soc-dummy";
```

Comment:
move at line 1057 and only allocate the codec_name in an else case then. No need to allocate and not use